### PR TITLE
Add backend https support 

### PR DESCRIPTION
**Pull request by fredead**
This changes the PROXY_SERVICE_HOST to include the protocol http or https
So that we can proxy https websites. Currently it is backwards compatible adding
http:// if https:// or http:// is not specified already present

While doing this commit I noted the change is not great as it assumes the protocol will be http of https.

### CODE REVIEW

File: enable_location.sh [hunk 1]
```diff
@@ -25,8 +25,13 @@ ERROR_REDIRECT_CODES=$(get_id_var ${LOCATION_ID} ERROR_REDIRECT_CODES)
 ENABLE_WEB_SOCKETS=$(get_id_var ${LOCATION_ID} ENABLE_WEB_SOCKETS)
 ADD_NGINX_LOCATION_CFG=$(get_id_var ${LOCATION_ID} ADD_NGINX_LOCATION_CFG)
 
+# Backwards compatability
+if [ "`echo ${PROXY_SERVICE_HOST} | grep -i 'https\?://'`" = "" ]; then
```
Comment (lewismarshall) [inline enable_location.sh]: Be good to comment this optional `https\?:` bash regexp scary :)

**Comment by lewismarshall** [comment]
Be good to add a backwards compatible test for no proto scheme specified...

**Comment by lewismarshall** [comment]
Chrz indeed though...

File: enable_location.sh (hunk 1 above)
Comment (fredead) [inline enable_location.sh]: Im not sure I like it all as it assumes http or https what if you proxied gopher or ftp ... 
i think we can just detect :// being missing although maybe this should just be another flag  ?

PROXY_HOST_INCLUDES_PROTOCOL=true ?

Thousght ?

Comment (lewismarshall) [inline enable_location.sh]: I like it transparent and default to http if not specified. We can always worry about other not http proxying in a separate PR...

**Comment by lewismarshall** [comment]
LGTM

**Pull request merged**